src/resources/order/spec/order.spec.js

```diff
@@ -396,3 +396,57 @@ describe('CANCEL order', () => {
     );
   });
 });
+
+describe('GET ALL orders', () => {
+  it('should fail if user is not authenticated', async () => {
+    const response = await request(app)
+      .get('/api/v1/orders')
+      .set({ Accept: 'application/json' });
+    expect(response.statusCode).toBe(401);
+    expect(response.body.error).toEqual(
+      'Access denied. You are not authorized to access this route',
+    );
+  });
+  it('should fail if user is not an admin', async () => {
+    const response = await request(app)
+      .get('/api/v1/orders')
+      .set({ 'x-auth-token': userToken, Accept: 'application/json' });
+    expect(response.statusCode).toBe(403);
+    expect(response.body.error).toEqual(
+      'You are not authorized to perform this action',
+    );
+  });
+  it('should fetch orders if user is admin with a limit', async () => {
+    const response = await request(app)
+      .get('/api/v1/orders')
+      .set({ 'x-auth-token': adminToken, Accept: 'application/json' });
+    expect(response.statusCode).toBe(200);
+  });
+  it('should be paginated', async () => {
+    const dbProducts = await db.raw('SELECT id FROM products');
+
+    const products = dbProducts.rows.map((product, idx) => {
+      const prod = {};
+      prod.id = product.id;
+      prod.quantity = idx + 1;
+      prod.total_price = 12.0 + idx * 2;
+      return prod;
+    });
+    const sampleOrder = {
+      shipping_address: 'No 3 my street, my city, ,my state',
+      shipping_fee: 4.99,
+      total_price: 12.99,
+    };
+    await request(app)
+      .post('/api/v1/orders')
+      .set({ 'x-auth-token': userToken, Accept: 'application/json' })
+      .send({ ...sampleOrder, products });
+
+    const response = await request(app)
+      .get('/api/v1/orders?page=1&pageSize=1')
+      .set({ 'x-auth-token': adminToken, Accept: 'application/json' });
+    expect(response.statusCode).toBe(200);
+    expect(response.body.message).toBe('2 Orders found');
+    expect(response.body.nextPage).toEqual(2);
```

Comment:
If limit is 10 and page one only returns two pages, there shouldn't be a next page